### DISCUSSION TITLE
Exemple of a query using the Microsoft search data source with the People entity

### DISCUSSION THREAD

**Discussion by coupalm** [microsoft-search/pnp-modern-search#2729] [Q&A]
Hi, I'm trying to build a people repository with the **Microsoft Search.** data source

My goals is to list all users and filter by office or department. I can do this with the SharePoint Data Source. 

Why using Microsoft Search? I want to retreive info in the AD profile that is not sync in SharePoint User Profile

In the search template this will work (return all users with "Coupal" in any properties)
{searchTerms} Coupal

If I do this: {searchTerms} or {searchTerms} *
There will be no result.

So my question is how can I get everyone? Note: If I go into OOTB Microsoft Search using the user scope and type *, it will display all users, so I would think it should do the same think in the search results...

Thanks for your help.

**Comment by kasperbolarsen** [comment]
Hi @coupalm I just checked the Graph Explorer and it seems like the people search option is in beta, and it requires a lot of Graph permissions. I am not sure if PnP Modern Search has those permissions, and if not then that might explain the issue.

**Reply by coupalm** [reply]
Hi @kasperbolarsen ,
I've just tested Graph beta and I get the same results as the search results web part. So what I'm looking for is how to get all persons in the directory....

The following returns nothing.
POST: https://graph.microsoft.com/beta/search/query
BODY
{
    "requests": [
        {
            "entityTypes": [
                "person"
            ],
            "query": {
                "queryString": "*",
                "queryTemplate": "({searchTerms})"
            }
        }
    ]
}

**Reply by kasperbolarsen** [reply]
I opened this issue with the Graph explorer today , https://github.com/microsoftgraph/microsoft-graph-explorer-v4/issues/2375

**Reply by coupalm** [reply]
Thanks!
